hal/hal_uv_packet.c

```diff
@@ -143,17 +143,18 @@ void adb_uv_on_data_available(adb_client_uv_t *client, uv_stream_t *stream,
     }
 
     if (nread == 0) {
-        /* No data available. This should not happen. FIXME */
-        adb_log("FIXME read returned no data.\n");
+        /* No data available. This should not happen. */
```

Comment:
This one is actually something to improve later on. Because it's reading exactly the size that was received, it's not possible from libuv to know if there is something else to read right after. So it would attempt another useless read just to be sure that there is no more data to receive, allocating apacket.
The proper fix would be to read size+1. Most of the time libuv would return size and wait for next packet to be received. if size +1 is returned then it means another packet arrived too.